builtin/providers/triton/provider.go

```diff
@@ -0,0 +1,130 @@
+package triton
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/joyent/gocommon/client"
+	"github.com/joyent/gosdc/cloudapi"
+	"github.com/joyent/gosign/auth"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"account": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SDC_ACCOUNT", ""),
+			},
+
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SDC_URL", "https://us-west-1.api.joyentcloud.com"),
+			},
+
+			"key_path": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SDC_KEY_PATH", ""),
+			},
+
+			"key_material": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SDC_KEY_MATERIAL", ""),
+			},
```

Comment:
So based on my follow up comment we should switch to key_material and drop key_path. We're trying to consistently accept values and let paths always be pulled in the same way - via `file()`.

Example of this change in the `google` provider, where we deprecated `account_file` and moved to `credentials`:

https://github.com/hashicorp/terraform/blob/9202bc0e1cf534e2c1992cb0c1c8358b0c195ef0/builtin/providers/google/provider.go#L16-L29

In acceptance tests we usually end up supporting a test-specific env var to suck a file from disk like so:

https://github.com/hashicorp/terraform/blob/9202bc0e1cf534e2c1992cb0c1c8358b0c195ef0/builtin/providers/google/provider_test.go#L33-L39

The reasoning for this pattern is to properly support build environments (Atlas, Jenkins, other CI) where a file on disk is not always easy to come by, but env vars are supported.

Let me know if this makes sense! :+1: